xdsl/transforms/lower_affine.py

```diff
@@ -29,34 +29,35 @@ def affine_expr_ops(
     Returns the operations that evaluate the affine expression when given input SSA
     values, along with the SSAValue representing the result.
     """
-    if isinstance(expr, AffineConstantExpr):
-        constant = arith.Constant(builtin.IntegerAttr.from_index_int_value(expr.value))
-        return [constant], constant.result
-
-    if isinstance(expr, AffineDimExpr):
-        return [], dims[expr.position]
-    if isinstance(expr, AffineSymExpr):
-        return [], symbols[expr.position]
-
-    if isinstance(expr, AffineBinaryOpExpr):
-        lhs_ops, lhs_val = affine_expr_ops(expr.lhs, dims, symbols)
-        rhs_ops, rhs_val = affine_expr_ops(expr.rhs, dims, symbols)
-
-        match expr.kind:
-            case AffineBinaryOpKind.Add:
-                op = arith.Addi(lhs_val, rhs_val)
-            case AffineBinaryOpKind.Mul:
-                op = arith.Muli(lhs_val, rhs_val)
-            case AffineBinaryOpKind.Mod:
-                op = arith.RemSI(lhs_val, rhs_val)
-            case AffineBinaryOpKind.FloorDiv:
-                op = arith.FloorDivSI(lhs_val, rhs_val)
-            case AffineBinaryOpKind.CeilDiv:
-                op = arith.CeilDivSI(lhs_val, rhs_val)
-
-        return [*lhs_ops, *rhs_ops, op], op.result
-
-    assert False, "Unreachable"
+    match expr:
+        case AffineConstantExpr():
```

Comment:
Is there a check that this actually works? Seems surprising to me that this does an isinstance check

Reply:
yep plenty of checks, this is the weird syntax python uses for isinstance checks in match statements